state/protocol/badger/state.go

```diff
@@ -42,7 +42,20 @@ func Bootstrap(
 	commits storage.EpochCommits,
 	statuses storage.EpochStatuses,
 	root protocol.Snapshot,
+	skipSnapshotValidation ...bool,
 ) (*State, error) {
+
+	// optionally skip validations involving flow identity
+	var skipValidations bool
+	switch len(skipSnapshotValidation) {
+	case 0:
+		skipValidations = false
+	case 1:
+		skipValidations = skipSnapshotValidation[0]
+	default:
+		return nil, fmt.Errorf("invalid value for skipSnapshotValidation")
+	}
```

Comment:
This looks really hacky. Can we either make this a functional option parameter, or just make it a required parameter?

Reply:
To use functional option, we would need to define a new struct

```golang
type StateConfig struct {
        skipValidations bool
        ...
}
```

Reply:
agreed..will change this